pkg/updatestatus/controlplaneinformer_test.go

```diff
@@ -0,0 +1,942 @@
+				Message: "Progressing=True in ClusterVersion but last history item has completion time",
+			},
+		},
+		{
+			name:          "Progressing=False, but last history item is not completed",
+			cvProgressing: &progressingFalse,
+			lastHistoryItem: &configv1.UpdateHistory{
+				State:       configv1.PartialUpdate,
+				StartedTime: minutesAgo[60],
+				Version:     "4.19.0",
+				Image:       "pullspec-4.19.0",
+			},
+			expectedCondition: metav1.Condition{
+				Status:  metav1.ConditionUnknown,
+				Reason:  "CannotDetermineUpdating",
+				Message: "Progressing=False in ClusterVersion but last history item is not completed",
+			},
+		},
+		{
+			name:          "Progressing=False, but last history item has no completion time",
+			cvProgressing: &progressingFalse,
+			lastHistoryItem: &configv1.UpdateHistory{
+				State:       configv1.CompletedUpdate,
+				StartedTime: minutesAgo[60],
+				Version:     "4.19.0",
+				Image:       "pullspec-4.19.0",
+			},
+			expectedCondition: metav1.Condition{
+				Status:  metav1.ConditionUnknown,
+				Reason:  "CannotDetermineUpdating",
+				Message: "Progressing=False in ClusterVersion but not no completion in last history item",
+			},
+		},
+		{
+			name: "Progressing=Unknown",
+			cvProgressing: &configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorProgressing,
+				Status:  configv1.ConditionUnknown,
+				Reason:  "WhoKnows",
+				Message: "Message about Progressing=Unknown",
+			},
+			lastHistoryItem: &partialUpdate,
+			expectedCondition: metav1.Condition{
+				Status:  metav1.ConditionUnknown,
+				Reason:  "CannotDetermineUpdating",
+				Message: "ClusterVersion has Progressing=Unknown(Reason=WhoKnows) | Message='Message about Progressing=Unknown'",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.expectedCondition.Type = string(ClusterVersionStatusInsightUpdating)
+
+			actualStatus, actualStarted, actualCompleted := isControlPlaneUpdating(tc.cvProgressing, tc.lastHistoryItem)
+			if diff := cmp.Diff(tc.expectedCondition, actualStatus); diff != "" {
+				t.Errorf("Status differs from expected:\n%s", diff)
+			}
+			if diff := cmp.Diff(tc.expectedStarted, actualStarted); diff != "" {
+				t.Errorf("Started differs from expected:\n%s", diff)
+			}
+			if diff := cmp.Diff(tc.expectedCompleted, actualCompleted); diff != "" {
+				t.Errorf("Completed differs from expected:\n%s", diff)
+			}
+		})
+	}
+}
+
+func Test_cvProgressingToUpdating(t *testing.T) {
+	testCases := []struct {
+		name        string
+		progressing configv1.ClusterOperatorStatusCondition
+
+		expectedStatus  metav1.ConditionStatus
+		expectedReason  string
+		expectedMessage string
+	}{
+		{
+			name: "Progressing=True",
+			progressing: configv1.ClusterOperatorStatusCondition{
+				Status:  configv1.ConditionTrue,
+				Reason:  "ReasonTrue",
+				Message: "MessageTrue",
+			},
+			expectedStatus: metav1.ConditionTrue,
+			// TODO(muller): Extract to API
```

Comment:
Could you please clarify what do we need to extract?

Reply:
Good catch. Nothing to extract now, I already did it. The API PR was missing these constants.